src/hacbs/utils/commits-utils.ts

```diff
@@ -62,3 +71,52 @@ export const getCommitsFromPLRs = (plrList: PipelineRunKind[], limit?: number):
   );
   return limit && limit < sortedCommits.length ? sortedCommits.slice(0, limit) : sortedCommits;
 };
+
+export const getCommitByName = (pipelineruns: PipelineRunKind[], commitName: string): Commit => {
```

Comment:
This function has duplicate code from `getCommitsFromPLRs`.

There should only be a single way to create a commit object.